Octokit/Http/JsonHttpPipeline.cs

```diff
@@ -48,12 +48,15 @@ public void DeserializeResponse<T>(IResponse<T> response)
             {
                 var body = response.Body;
                 // simple json does not support the root node being empty. Will submit a pr but in the mean time....
-                if (body != "{}") 
+                if (body != "{}")
                 {
+                    var typeIsDictionary = typeof(IDictionary).IsAssignableFrom(typeof(T));
+                    var typeIsEnumerable = typeof(IEnumerable).IsAssignableFrom(typeof(T));
+                    var responseIsArray = body.StartsWith("{", StringComparison.Ordinal);
+
                     // If we're expecting an array, but we get a single object, just wrap it.
                     // This supports an api that dynamically changes the return type based on the content.
-                    if ((typeof(IEnumerable).IsAssignableFrom(typeof(T)))
```

Comment:
`IDictionary` subclasses `IEnumerable`, so some of our tests which return `Dictionary<T1,T2>` were being caught here and deserialized incorrectly.

Feel free to submit a PR to make this a bit more cleaner.